users/views.py

```diff
@@ -39,23 +33,34 @@ def get(self, request, *args, **kwargs):
         return super(UserRegisterView, self).get(request, *args, **kwargs)
 
 
-class UserUpdateView(LoginRequiredMixin, UserPassesTestMixin, UpdateView):
+class UserDetailView(LoginRequiredMixin, UserPassesTestMixin, DetailView):
     template_name = 'users/profile.html'
     model = get_user_model()
+
+    def test_func(self):
+        """
+        this func check that the user which want
+        to update profile should have permission to only his profile
+        """
+        current_user = self.get_object()
+
+        return self.request.user == current_user
+
+
+class UserUpdateView(LoginRequiredMixin, UserPassesTestMixin, UpdateView):
+    template_name = 'users/update_profile.html'
+    model = get_user_model()
     form_class = UserUpdateForm
     success_url = '/'
```

Comment:
In your code this line is useless, i can to propose something like:
```suggestion
def get_success_url(self):
    """ Invoked after call form_valid() method """
    return reverse_lazy('profile', kwargs={'pk': self.object.pk})
```